src/hooks/govwallet/useGovWalletBalance/useGovWalletBalance.test.tsx

```diff
@@ -18,266 +18,323 @@ const mockGetGovWalletBalance = getGovWalletBalance as jest.Mock;
-    expect.assertions(3);
-
-    mockGetGovWalletBalance.mockResolvedValueOnce({
-      accountDetails: [
-        {
-          activationStatus,
-          balance: 10000,
-          modified: "2022-07-25T11:57:50.000+08:00",
-        },
-      ],
+    let ids: string[];
+
+    describe("isPayNowTransaction toggle", () => {
+      beforeAll(() => {
+        ids = ["S0000000J"];
+
+        mockGetGovWalletBalance.mockResolvedValue({
+          accountDetails: [
+            {
+              activationStatus: "ACTIVATED",
+              balance: 10000,
+              modified: "2022-07-25T11:57:50.000+08:00",
+            },
+          ],
+        });
+      });
+
+      it("should get balance if toggle is true", async () => {
+        expect.assertions(3);
+
+        const { result, waitForNextUpdate } = renderHook(
+          () => useGovWalletBalance(ids, key, endpoint),
+          { wrapper }
+        );
+
+        expect(result.current.govWalletBalanceState).toStrictEqual(
+          "FETCHING_BALANCE"
+        );
+        await waitForNextUpdate();
+        expect(result.current.govWalletBalanceState).toStrictEqual("ELIGIBLE");
+        expect(mockGetGovWalletBalance).toHaveBeenCalledTimes(1);
+      });
+
+      it("should NOT get balance if toggle is false", async () => {
+        expect.assertions(2);
+
+        const { result } = renderHook(
+          () => useGovWalletBalance(ids, key, endpoint),
+          { wrapper: wrapperWithToggleOff }
+        );
+
+        expect(result.current.govWalletBalanceState).toStrictEqual("DEFAULT");
+        expect(mockGetGovWalletBalance).toHaveBeenCalledTimes(0);
+      });
     });
 
-    const { result, waitForNextUpdate } = renderHook(
-      () => useGovWalletBalance(ids, key, endpoint),
-      { wrapper }
-    );
-
-    expect(result.current.govWalletBalanceState).toStrictEqual(
-      "FETCHING_BALANCE"
-    );
-    await waitForNextUpdate();
-    expect(result.current.govWalletBalanceState).toStrictEqual("ELIGIBLE");
-    expect(mockGetGovWalletBalance).toHaveBeenCalledTimes(1);
-  });
-
-  it("should return 'INELIGIBLE' if balance is more than 10000 cents", async () => {
-    expect.assertions(3);
-
-    mockGetGovWalletBalance.mockResolvedValueOnce({
-      accountDetails: [
-        {
-          activationStatus,
-          balance: 10001,
-          modified: "2022-07-25T11:57:50.000+08:00",
-        },
-      ],
+    describe("govWalletBalanceState states", () => {
+      let activationStatus: string;
+
+      beforeAll(() => {
+        ids = ["S0000000J"];
+        activationStatus = "ACTIVATED";
+      });
+
+      beforeEach(() => {
+        jest.clearAllMocks();
+      });
+
+      it("should return 'ELIGIBLE' if balance is strictly equal to 10000 cents", async () => {
+        expect.assertions(3);
+
+        mockGetGovWalletBalance.mockResolvedValueOnce({
+          accountDetails: [
+            {
+              activationStatus,
+              balance: 10000,
```

Comment:
Can replace all instances in the tests where this is hardcoded to `defaultFeatures.govwalletExactBalanceValue`?

Reply:
updated~